lib/include/ert/nexus/nexus_plot.hpp

```diff
@@ -39,25 +44,270 @@ struct unexpected_eof : public std::runtime_error {
     using std::runtime_error::runtime_error;
 };
 
-class NexusPlot {
-public:
-    NexusPlot( const std::string& );
-    NexusPlot( std::istream& );
-
-    int32_t num_classes = 0;
+struct NexusHeader {
+    int32_t num_classes;
     int32_t day, month, year;
     int32_t nx, ny, nz;
     int32_t ncomp;
-    std::vector<std::string> class_names;
-    std::vector<int32_t> vars_in_class;
-    std::vector< std::vector<std::string> > var_names;
+};
+
 
-    ecl_sum_type* ecl_summary( const std::string& ecl_case );
+struct NexusData {
+    int32_t timestep;
+    float time;
+    int32_t max_perfs;
+    std::array< char, 8 > classname;
```

Comment:
Why not just use `std::string` - if this is a performance optimization - *it is not worth it*.

Reply:
It is both for performance reasons and the fact that the we know that all classnames and instancenames are 8 characters long. This makes construction simple, and comparisons can be done without having to worry about null-terminators. Strings can, but does not need to, contain null-terminators. When comparing two equal strings, one with null-terminator and one without, c++ will say they are different. It is also easier to read into an std::array as std::array::data is guaranteed to return a sequential pointer, std::string is not (until c++14).